src/types.ts

```diff
@@ -1,10 +1,73 @@
-import { MouseEvent } from 'react'
-import { ItemProps } from './item'
+import { MouseEvent, MutableRefObject, ReactNode } from 'react'
+import PhotoSwipe, { PhotoSwipeOptions, UIElementData } from 'photoswipe'
+import PhotoSwipeLightboxStub from './lightbox-stub'
 
-export type ItemRef = React.MutableRefObject<HTMLElement>
+/**
+ * At Gallery level we can freely assume that ref is HTMLElement since we don't use any of html attributes.
+ * At Item level we can either set ref type like <Item<HTMLImageElement>> to ensure typing or simply omit it since
+ * elements like HTMLImageElement extends HTMLElement.
+ */
+export type ItemRef<NodeType extends HTMLElement = HTMLElement> =
+  MutableRefObject<NodeType | null>
 
-export type InternalItem = Omit<ItemProps, 'children'>
+export interface InternalItem {
```

Comment:
and probably we should not export to public `InternalItem` props

Reply:
I think the best practice is to export user-land types manually. It prevents from exporting some internal api. Also we can see what types are actually available from outside so we can't accidentally remove some export like `ItemProps` (like i did)

Reply:
@MaximKondratev, nice idea🔥